packages/client/src/components/TransferToken/steps/DepositTokenForm.js

```diff
@@ -0,0 +1,106 @@
+import { useContext, useEffect, useState } from 'react';
+import { Web3Context } from 'contexts/Web3';
+import { TransferTokensContext } from 'components/TransferToken/TransferTokensContext';
+import AmountInput from 'components/TransferToken/components/AmountInput';
+import AssetSelector from 'components/TransferToken/components/AssetSelector';
+import { useAccount, useContacts } from 'hooks';
+import { ASSET_TYPES } from 'constants/enums';
+import { formatAddress, getNameByAddress, isVaultInTreasury } from 'utils';
+import Svg from 'library/Svg';
+
+const DepositTokenForm = () => {
+  const [coinBalances, setCoinBalances] = useState();
+  const { getUserBalances } = useAccount();
+
+  const [sendModalState] = useContext(TransferTokensContext);
+  const {
+    assetType,
+    sender: userAddress,
+    recipient: safeAddress,
+  } = sendModalState;
+
+  const web3 = useContext(Web3Context);
+  const { name: safeName } = web3?.treasuries?.[safeAddress];
+  const { getTreasuryVaults, vaults } = web3;
+  const { contacts } = useContacts(safeAddress);
+
+  const getVaults = async () => {
+    await getTreasuryVaults(safeAddress);
+  };
+
+  const updateUserBalance = async () => {
+    try {
+      const userBalances = await getUserBalances(formatAddress(userAddress));
+      const balances = {};
+      userBalances.forEach((userBalance) => {
+        const { coinType, balance } = userBalance;
+        if (isVaultInTreasury(vaults[safeAddress], coinType)) {
+          balances[userBalance.coinType] = balance;
+        }
+      });
+
+      setCoinBalances(balances);
+    } catch (err) {
+      console.log(`Failed to get coin balances, error: ${err}`);
+    }
+  };
+
+  useEffect(() => {
+    if (!userAddress) {
+      return;
+    }
+    getVaults();
+    // eslint-disable-next-line
```

Comment:
```suggestion
   // eslint-disable-next-line react-hooks/exhaustive-deps
```

we should probably get out of the habit of abusing this eslint ignore though, apparently its not _good code_ https://dev.to/wkrueger/never-ignore-the-exhaustive-deps-rule-2ap8